# Explore here

In [10]:
import pandas as pd
import os
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error, r2_score
import scipy.stats as stats

# Planteamiento del problema y recolección de la data

* Una empresa ha recolectado la información del alquiler de viviendas en Nueva York a través de la aplicación Airbnb durante el año 2019
* El objetivo principal de este analisis estara enfocado en establecer en que vecindarios se presentan los mayores valores de los inmuebles.

In [15]:
# Lectura y guardado data
# URL del archivo CSV
url = 'https://raw.githubusercontent.com/4GeeksAcademy/linear-regression-project-tutorial/main/medical_insurance_cost.csv'
df_mi=pd.read_csv('https://raw.githubusercontent.com/4GeeksAcademy/linear-regression-project-tutorial/main/medical_insurance_cost.csv')

In [16]:
# Guardado de la data
df_mi = pd.read_csv(url)
df_mi_inicial = pd.read_csv(url)


# Ruta de destino
ruta_destino = '/workspaces/Proyecto-de-Regresi-n-Lineal/data/raw/medical_insurance_cost.csv'
ruta_destino_i = '/workspaces/Proyecto-de-Regresi-n-Lineal/data/raw/medical_insurance_cost_inicial.csv'

# Crear el directorio si no existe
os.makedirs(os.path.dirname(ruta_destino), exist_ok=True)

# Guardar el DataFrame en la ruta especificada con una copia de la data original
df_mi.to_csv(ruta_destino, index=False)
df_mi_inicial.to_csv(ruta_destino_i, index=False)

print(f'Archivo guardado en: {ruta_destino}')

Archivo guardado en: /workspaces/Proyecto-de-Regresi-n-Lineal/data/raw/medical_insurance_cost.csv


In [17]:
df_mi.head()

age     sex     bmi  children smoker     region      charges
0   19  female  27.900         0    yes  southwest  16884.92400
1   18    male  33.770         1     no  southeast   1725.55230
2   28    male  33.000         3     no  southeast   4449.46200
3   33    male  22.705         0     no  northwest  21984.47061
4   32    male  28.880         0     no  northwest   3866.85520

In [18]:
df_mi.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1338 entries, 0 to 1337
Data columns (total 7 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   age       1338 non-null   int64  
 1   sex       1338 non-null   object 
 2   bmi       1338 non-null   float64
 3   children  1338 non-null   int64  
 4   smoker    1338 non-null   object 
 5   region    1338 non-null   object 
 6   charges   1338 non-null   float64
dtypes: float64(2), int64(2), object(3)
memory usage: 73.3+ KB


In [19]:
df_mi.describe()

age          bmi     children       charges
count  1338.000000  1338.000000  1338.000000   1338.000000
mean     39.207025    30.663397     1.094918  13270.422265
std      14.049960     6.098187     1.205493  12110.011237
min      18.000000    15.960000     0.000000   1121.873900
25%      27.000000    26.296250     0.000000   4740.287150
50%      39.000000    30.400000     1.000000   9382.033000
75%      51.000000    34.693750     2.000000  16639.912515
max      64.000000    53.130000     5.000000  63770.428010

In [20]:
df_mi.columns

Index(['age', 'sex', 'bmi', 'children', 'smoker', 'region', 'charges'], dtype='object')

# Exploración y limpieza de los Datos

In [22]:
# Valores Nulos
df_mi.isnull().sum()

age         0
sex         0
bmi         0
children    0
smoker      0
region      0
charges     0
dtype: int64

In [25]:
# Eliminamos duplicados
dp=df_mi[df_mi.duplicated(keep=False)]
df_mi.drop_duplicates(inplace=True)

El Dataset esta conformado por 1337 valores, 7 variables, incluyendo la clase a predecir (charges), las predictoras estan conformadas por: age, sex, bmi, children, smoker, region. No tiene valores nulos, solo tiene un registro duplicado y procedemos a eliminarlo.


La distribucion de los tipos de variables es la siguiente:

* age int64
* sex object
* bmi float64
* children int64
* smoker object
* region object
* charges float64

Con el analisis inicial de variables del Dataset las variables region y children no tienen un peso importante en la clase a predecir y por tanto se procede a eliminarlas.

In [24]:
df_mi.drop(axis=1,columns=['region','children'],inplace=True)
df_mi.columns

Index(['age', 'sex', 'bmi', 'smoker', 'charges'], dtype='object')

# Análisis de variables

Variables univariante